relay-profiling/src/sample.rs

```diff
@@ -232,20 +232,28 @@ fn parse_profile(payload: &[u8]) -> Result<SampleProfile, ProfileError> {
         return Err(ProfileError::MissingProfileMetadata);
     }
 
-    let Some(transaction) = profile.transactions.get(0).cloned() else {
-        return Err(ProfileError::NoTransactionAssociated);
-    };
+    if profile.transaction.is_none() {
+        profile.transaction = profile.transactions.first().cloned();
```

Comment:
I overlooked that `Vec::take` is nightly API, but this (though admittedly ugly) should work:
```suggestion
        profile.transaction = std::mem::take(&mut profile.transactions).into_iter().next();
```

Reply:
That's the same as `profile.transactions.drain(..)...`